htdocs/core/modules/facture/doc/pdf_crabe.modules.php

```diff
@@ -43,11 +43,16 @@
  */
 class pdf_crabe extends ModelePDFFactures
 {
-	 /**
-	  * @var DoliDb Database handler
-	  */
+	/**
+	 * @var DoliDb Database handler
+	 */
 	public $db;
 
+	/**
+	 * @var int The environment ID when using a multicompany module
+	 */
+	public $entity;
```

Comment:
Why adding an entity field on a PDF document ? code to generate a document does not depends on entity. Also this property is not used in code of PDF generators.

Reply:
Line 345 👍 $pagecount = $pdf->setSourceFile($conf->mycompany->multidir_output[$object->entity].'/'.$conf->global->MAIN_ADD_PDF_BACKGROUND); ?

Reply:
Yes entity exists on $object, not on $this which is a pdf class